p2p/tests/buckets.rs

```diff
@@ -4,6 +4,71 @@ use witnet_p2p::peers::{
     calculate_index_for_new, calculate_index_for_tried, split_socket_addresses,
 };
 
+/// Tests for the business logic of inserting and removing peer addresses into the `ice` bucket.
+mod ice {
+    use super::ip;
+    use std::time::Duration;
+    use witnet_p2p::peers::Peers;
+
+    #[test]
+    fn test_can_ice_an_address() {
+        let mut peers = Peers::new();
+        peers.set_ice_period(Duration::from_secs(1000));
+        let address = ip("192.168.1.1:21337");
+        let can_ice = peers.ice_peer_address(&address);
+
+        assert!(can_ice);
+
+        let is_iced = peers.ice_bucket_contains(&address);
+
+        assert!(is_iced)
+    }
+
+    #[test]
+    fn test_icing_blocks_the_entire_ip() {
```

Comment:
What's the reasoning behind this?

So, assuming that an ip is "iced" when there is no node running there, if I run two nodes on the same ip but later stop one, there is a chance that a new node will try to connect to the old node address, see that there is no node running there, and block the other perfectly valid node?

Also, an attacker can broadcast peers with an invalid port to move that ip into the ice bucket, essentially banning a node from connecting to certain ips. So if there is a node running on 192.168.1.1:21337, an attacker can broadcast 192.168.1.1:1 as a peer address, and other nodes will try to connect, see that there is no node there, and "ice" this ip.

Also, assuming that nodes running on different consensus constants are "iced", this may interfere with private testnets: 2 nodes running in different testnets in the same ip may cause other nodes to block this ip range.

Reply:
True, the implications of this are quite serious. Will modify it so it only ices the specific port.